browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -173,6 +202,30 @@ void SpeedreaderTabHelper::HideBubble() {
   }
 }
 
+void SpeedreaderTabHelper::OnDistillComplete() {
+  // Perform a state transition
+  if (distill_state_ == DistillState::kSpeedreaderModePending) {
+    distill_state_ = DistillState::kSpeedreaderMode;
+  } else if (distill_state_ == DistillState::kReaderModePending) {
+    distill_state_ = DistillState::kReaderMode;
+  } else {
+    // We got here via an already cached page.
+    DCHECK(distill_state_ == DistillState::kSpeedreaderMode ||
+           distill_state_ == DistillState::kReaderMode);
+  }
+}
+
+void SpeedreaderTabHelper::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle->HasCommitted()) {
+    distill_state_ = DistillState::kNone;
+  }
+  auto* entry = web_contents()->GetController().GetLastCommittedEntry();
+  DCHECK(entry);
```

Comment:
I checked other code and it seems we'd better have `if`

Reply:
fixed and also moved to `NavigationEntryCommitted`